scripts/hgsc_subtypes_tybalt.py

```diff
@@ -0,0 +1,275 @@
+                              how='right', left_index=True, right_index=True)
+ov_encoded_subtype = ov_encoded_subtype.assign(subtype_color = ov_encoded_subtype['SUBTYPE'])
+
+ov_subtype_color_dict = {'Differentiated': 'purple',
+                         'Immunoreactive': 'green',
+                         'Mesenchymal': 'blue',
+                         'Proliferative': 'red'}
+ov_encoded_subtype = ov_encoded_subtype.replace({'subtype_color': ov_subtype_color_dict})
+
+print(ov_encoded_subtype.shape)
+ov_encoded_subtype.head(2)
+
+
+# In[9]:
+
+# Get the HGSC mean feature activation
+ov_mean_subtypes = ov_encoded_subtype.groupby('SUBTYPE').mean()
+ov_mean_subtypes
+
+
+# ## HGSC Subtype Math
+# 
+# Because of the relationship observed in the consistent clustering solutions, perform the following subtractions
+# 
+# 1. Immunoreactive - Mesenchymal
+# 2. Differentiated - Proliferative
+# 
+# The goal is to observe the features with the largest difference between the aformentioned comparisons. The differences should be in absolute directions
+
+# In[10]:
+
+mes_mean_vector = ov_mean_subtypes.loc['Mesenchymal', [str(x) for x in range(1, 101)]]
+imm_mean_vector = ov_mean_subtypes.loc['Immunoreactive', [str(x) for x in range(1, 101)]]
+pro_mean_vector = ov_mean_subtypes.loc['Proliferative', [str(x) for x in range(1, 101)]]
+dif_mean_vector = ov_mean_subtypes.loc['Differentiated', [str(x) for x in range(1, 101)]]
+
+
+# In[11]:
+
+high_immuno = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).head(2)
+high_mesenc = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).tail(2)
+
+print("Features with large differences: Immuno high, Mesenchymal low")
+print(high_immuno)
+print("Features with large differences: Mesenchymal high, Immuno low")
+print(high_mesenc)
+
+
+# In[12]:
+
+# Select to visualize encoding 56 because it has high immuno and low everything else
+ov_mean_subtypes.loc[:, ['87', '77', '56']]
+
+
+# In[13]:
+
+# Obtain the decoder weights
+weights = []
+for l in decoder.layers:
+    weights.append(l.get_weights())
+    
+weight_layer = pd.DataFrame(weights[1][0], columns=rnaseq_df.columns)
+weight_layer.head(2)
+
+
+# In[14]:
+
+# Get the high weight genes describing the immunoreactive subtype
+i_87_genes = weight_layer.loc[87, :].sort_values(ascending = False)
+i_77_genes = weight_layer.loc[77, :].sort_values(ascending = False)
+i_56_genes = weight_layer.loc[56, :].sort_values(ascending = False)
+
+
+# In[15]:
+
+# File names for output genes
+node87pos_file = os.path.join('results', 'hgsc_node87genes_pos.tsv')
+node87neg_file = os.path.join('results', 'hgsc_node87genes_neg.tsv')
+node77pos_file = os.path.join('results', 'hgsc_node77genes_pos.tsv')
+node77neg_file = os.path.join('results', 'hgsc_node77genes_neg.tsv')
+node56pos_file = os.path.join('results', 'hgsc_node56genes_pos.tsv')
+node56neg_file = os.path.join('results', 'hgsc_node56genes_neg.tsv')
+
+high_std = 2
+col_names = ['gene', 'weight']
```

Comment:
This is unused. Did you mean to add this as column headers in the to_csv functions below?

Reply:
yikes, yes I did. Thanks

Reply:
updated in commit c389fa3